### PR TITLE
Add Interaction events and data object

### PR DESCRIPTION
**Pull request by ReveredRedHood**

### CODE REVIEW

**Comment by ReveredRedHood** [comment]
Note that the extra .dll references added to .csproj may or may not be needed by this particular change. However, they will definitely be needed for future changes involving TextMeshPro

**Comment by ReveredRedHood** [comment]
I need to document the events, will do soon

**Comment by Ven0maus** [comment]
#8

File: SOD.Common/Helpers/Interaction.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+namespace SOD.Common.Helpers
+{
+    public sealed class Interaction
+    {
+        private float _lastAmount = float.MaxValue;
+        internal SimpleActionArgs CurrentPlayerInteraction;
+        internal bool LongActionInProgress = false;
+
+        public event EventHandler<SimpleActionArgs> OnBeforeActionStarted;
```
Comment (Ven0maus) [inline SOD.Common/Helpers/Interaction.cs:12]: And short descriptive comment on each event

File: SOD.Common/Custom/InteractableInstanceData.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+using UniverseLib.Utility;
+
+namespace SOD.Common.Custom
+{
+    /// <summary>
+    /// Provides safe access to Interactable data, including TryGet methods for InteractableControllers and Unity objects. Use this to avoid issues when Interactables get pooled.
+    /// </summary>
+    public sealed class InteractableInstanceData
+    {
+        public InteractableInstanceData(Interactable interactable, InteractablePreset preset)
```
Comment (Ven0maus) [inline SOD.Common/Custom/InteractableInstanceData.cs:11]: make internal

**Pull request merged**